doc/dev/encoding.rst

```diff
@@ -30,36 +30,34 @@ by a programmer by implementing the ``encode`` and ``decode`` methods.
 
 Principles for format change
 ----------------------------
-It is not unusual that the format of serialization changes. This
-process requires careful attention during both development
+It is not unusual for the format of serialization to change. This
+process requires careful attention both during development
 and review.
 
-The general rule is that a decoder must understand what had been
-encoded by an encoder. Most of the problems come from ensuring
-that compatibility continues between old decoders and new encoders
-as well as new decoders and old decoders. One should assume
-that -- if not otherwise derogated -- any mix (old/new) is
-possible in a cluster. There are 2 main reasons for that:
-
-1. Upgrades. Although there are recommendations related to the order
-   of entity types (mons/osds/clients), it is not mandatory and
-   no assumption should be made about it.
-2. Huge variability of client versions. It was always the case
-   that kernel (and thus kernel clients) upgrades are decoupled
-   from Ceph upgrades. Moreover, proliferation of containerization
-   bring the variability even to e.g. ``librbd`` -- now user space
-   libraries live on the container own.
-
-With this being said, there are few rules limiting the degree
-of interoperability between dencoders:
+The general rule is that a decoder must understand what has been encoded by an
+encoder. Most difficulties arise during the process of ensuring the continuity
+of compatibility of old decoders with new encoders, and ensuring the continuity
+of compatibility of new decoders with old decoders. One should assume -- if not
+otherwise specified -- that any mix of old and new is possible in a cluster.
+There are 2 main reasons for this:
+
+1. **Upgrades.** Although there are recommendations related to the order of
+   entity types (mons/osds/clients), it is not mandatory and no assumption
+   should be made about it.
+2. **Huge variability of client versions.** It was always the case that kernel
+   upgrades (and thus kernel clients) are decoupled from Ceph upgrades.
+   Containerization brings the variability even to e.g. ``librbd`` -- now user
+   space libraries live on the container itself:
+
+There are few rules limiting the degree of interoperability between dencoders:
```

Comment:
>We enforce the following for incompatible changes:

As written this implies that interoperability must not be maintained.

Reply:
I don't know where the quoted text ("We enforce the following for incompatible changes") is, but I can see that "There are few rules limiting the degree of interoperability between dencoders" runs into the few/a few subtlety whereby "There are few tickets left" means that there are almost no tickets left and (depending on how much understatement is being used by the reporter) there are possibly none, but "There are _a_ few tickets left" means that there are indeed tickets left.

I've changed 
`There are few rules limiting the degree of interoperability between dencoders:`
to read
`There are a few rules limiting the degree of interoperability between dencoders:`